hydra-cluster/src/Hydra/Cluster/Faucet.hs

```diff
@@ -126,6 +109,58 @@ seedFromFaucet_ ::
 seedFromFaucet_ node vk ll marked tracer =
   void $ seedFromFaucet node vk ll marked tracer
 
+-- | Return the remaining funds to the faucet
+returnFundsToFaucet ::
+  Tracer IO FaucetLog ->
+  RunningNode ->
+  Actor ->
+  IO ()
+returnFundsToFaucet tracer RunningNode{networkId, nodeSocket} sender = do
+  (faucetVk, _) <- keysFor Faucet
+  let faucetAddress = mkVkAddress networkId faucetVk
+
+  (senderVk, senderSk) <- keysFor sender
+  utxo <- queryUTxOFor networkId nodeSocket QueryTip senderVk
+
+  retryOnExceptions tracer $ do
+    let allLovelace = selectLovelace $ balance @Tx utxo
+    -- XXX: Using a hard-coded high-enough value to satisfy the min utxo value.
+    -- NOTE: We use the faucet address as the change deliberately here.
+    tx <- sign senderSk <$> buildTxBody utxo faucetAddress 1_000_000
+    let fee = txFee' tx
+    let returnBalance = allLovelace - fee
+    tx' <- sign senderSk <$> buildTxBody utxo faucetAddress returnBalance
+    submitTransaction networkId nodeSocket tx'
+    void $ awaitTransaction networkId nodeSocket tx'
+    traceWith tracer $ ReturningFunds{actor = actorName sender, returnAmount = returnBalance}
```

Comment:
Could: use past tense to indicate this already happened, i.e. name it `ReturnedFunds`